core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -17,247 +17,342 @@
-        return explorationMode === EXPLORATION_MODE.PRETEST;
-      },
-      isInQuestionMode: function() {
-        return explorationMode === EXPLORATION_MODE.PRETEST ||
-        explorationMode === EXPLORATION_MODE.QUESTION_PLAYER;
-      },
-      isInQuestionPlayerMode: function() {
-        return explorationMode === EXPLORATION_MODE.QUESTION_PLAYER;
-      },
-      isInStoryChapterMode: function() {
-        return explorationMode === EXPLORATION_MODE.STORY_CHAPTER;
-      },
-      getPretestQuestionCount: function() {
-        return QuestionPlayerEngineService.getPretestQuestionCount();
-      },
-      moveToExploration: function(callback) {
-        if (
-          UrlService.getUrlParams().hasOwnProperty('story_url_fragment') &&
-          UrlService.getUrlParams().hasOwnProperty('node_id')) {
-          setStoryChapterMode();
-        } else {
-          setExplorationMode();
-        }
-        ExplorationEngineService.moveToExploration(callback);
-      },
-      getLanguageCode: function() {
-        return currentEngineService.getLanguageCode();
-      },
-      recordNewCardAdded: function() {
-        return currentEngineService.recordNewCardAdded();
-      },
-      get onTotalQuestionsReceived() {
-        return _totalQuestionsReceivedEventEmitter;
-      },
-      get onPlayerStateChange() {
-        return _playerStateChangeEventEmitter;
+    this.storyUrlFragment = this.urlService.getStoryUrlFragmentFromLearnerUrl();
+  }
+
+  initializeExplorationServices(
+      returnDict: FetchExplorationBackendResponse,
+      arePretestsAvailable: boolean,
+      callback: (stateCard: StateCard, str: string) => void
+  ): void {
+    // For some cases, version is set only after
+    // ReadOnlyExplorationBackendApiService.loadExploration() has completed.
+    // Use returnDict.version for non-null version value.
+    this.statsReportingService.initSession(
+      this.explorationId, returnDict.exploration.title, returnDict.version,
+      returnDict.session_id,
+      this.urlService.getCollectionIdFromExplorationUrl());
+    this.playthroughService.initSession(
+      this.explorationId, returnDict.version,
+      returnDict.record_playthrough_probability);
+    this.playerCorrectnessFeedbackEnabledService.init(
+      returnDict.correctness_feedback_enabled);
+    this.explorationEngineService.init(
+      {
+        auto_tts_enabled: returnDict.auto_tts_enabled,
+        correctness_feedback_enabled: returnDict.correctness_feedback_enabled,
+        draft_changes: [],
+        is_version_of_draft_valid: true,
+        init_state_name: returnDict.exploration.init_state_name,
+        param_changes: returnDict.exploration.param_changes,
+        param_specs: returnDict.exploration.param_specs,
+        states: returnDict.exploration.states,
+        title: returnDict.exploration.title,
+        language_code: returnDict.exploration.language_code,
+        version: returnDict.version
       },
```

Comment:
Converting
this (returnDict):
![Screenshot from 2021-05-17 11-40-55](https://user-images.githubusercontent.com/24855641/118447063-f6eb3680-b70d-11eb-8a76-d566ab3550e9.png)

to:
![Screenshot from 2021-05-17 12-09-32](https://user-images.githubusercontent.com/24855641/118447099-01a5cb80-b70e-11eb-8971-337e9ca78767.png)